src/Raven.Server/Documents/Handlers/Admin/RachisAdminHandler.cs

```diff
@@ -265,6 +264,87 @@ public Task EnforceReelection()
             return Task.CompletedTask;
         }
 
+        /* Promote a non-voter to a promotable */         
+        [RavenAction("/admin/cluster/promote", "POST", "/admin/cluster/promote?nodeTag={nodeTag:string}")]
+        public async Task PromoteNode()
+        {
+            if (ServerStore.LeaderTag == null)
+                return;
+            
+            if (ServerStore.IsLeader() == false)
+            {
+                RedirectToLeader();
+                return;
+            }
+
+            var nodeTag = GetStringQueryString("nodeTag");
+            using (ServerStore.ContextPool.AllocateOperationContext(out TransactionOperationContext context))
+            using (context.OpenReadTransaction())
+            {
+                var topology = ServerStore.GetClusterTopology(context);
+                if (topology.AllNodes.ContainsKey(nodeTag) == false)
+                {
+                    throw new InvalidOperationException(
+                        $"Failed to promote node {nodeTag} beacuse {nodeTag} is not a part of the cluster topology");
+                }
+                if (topology.Watchers.ContainsKey(nodeTag) == false)
+                {
+                    throw new InvalidOperationException(
+                        $"Failed to promote node {nodeTag} beacuse {nodeTag} is already a voter in the cluster topology");
+                }
+
+                var url = topology.GetUrlFromTag(nodeTag);
+                var publicKey = topology.GetPublicKeyFromTag(nodeTag);
+                await ServerStore.Engine.ModifyTopologyAsync(nodeTag, url, Leader.TopologyModification.Promotable, publicKey);
+                HttpContext.Response.StatusCode = (int)HttpStatusCode.OK;
+            }
+        }
+
+        /* Demote a voter (member/promotable) node to a non-voter  */
+        [RavenAction("/admin/cluster/demote", "POST", "/admin/cluster/demote?nodeTag={nodeTag:string}")]
+        public async Task DemoteNode()
+        {
+            if (ServerStore.LeaderTag == null)            
+                return;           
+            if (ServerStore.IsLeader() == false)
+            {
+                RedirectToLeader();
+                return;
+            }
+
+            var nodeTag = GetStringQueryString("nodeTag");
+            if (nodeTag == ServerStore.LeaderTag)
```

Comment:
I think it is better if we won't do this on our own, and just have the admin do it.
Refusing to demote a leader is easier.